main/src/cgeo/geocaching/ImageViewActivity.java

```diff
@@ -146,7 +148,7 @@ public void setPrimaryItem(@NonNull final ViewGroup container, final int positio
 
         @Override
         public int getCount() {
-            return realImageSize * ENDLESS_MULTIPLIER;
+            return realImageSize == 1 ? 1 : realImageSize * ENDLESS_MULTIPLIER;
```

Comment:
Not that I am able to do a full review here, I did however stumble over this line, which looks a bit unrelated to the topic. Just want to make sure nothing unwanted slipped into this PR?!

Reply:
You have good eyes. This line is not strictly related to the PR at hand, but it fixes another problem which has bothered me:

as you know the image view implements "endless swipe", meaning that if you are are the first image and swipe right then you come to the last image again and the other way around.

However this behavior does not make sense when there's only a single image in the list. The change in the line thua deactivates swiping if there's only a single image.

Reply:
Thumbs up